commcare/src/main/java/org/motechproject/commcare/pull/CommcareTasksFormImporterFactory.java

```diff
@@ -0,0 +1,55 @@
+package org.motechproject.commcare.pull;
+
+import org.motechproject.commcare.service.CommcareFormService;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.EventRelay;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is the factory responsible for providing instances {@link CommcareFormImporter} to
+ * import form event handling {@link org.motechproject.commcare.events.imports.ImportFormActionEventHandler}
+ */
+@Component
+public class CommcareTasksFormImporterFactory {
+
+    @Autowired
+    private EventRelay eventRelay;
+
+    @Autowired
+    private CommcareFormService formService;
+
+    private final Map<String, CommcareFormImporterImpl> importerMap = new HashMap<>();
```

Comment:
We do not need that map. Just make the method return a new instance of CommcareFormImporterImpl